### PR TITLE
Disable node/proxy functionality in Isolated mode

### LINKED ISSUE #937: Disable node/proxy functionality in Isolated mode and add a flag to optionally enable it
### Is your feature request related to a problem?

The `node/proxy` functionality in Kubernetes can potentially allow circumvention of the security measures introduced by the Isolated mode of vcluster. AFAIK it is mitigated when "fake nodes" are used, but not with the "real nodes" sync.

### Which solution do you suggest?

The `node/proxy` RBAC permission should not be added to the ClusterRole in the Helm template whenever isolated mode is enabled.
A new Helm value should be added to optionally allow original behavior even in the isolated mode.

### Which alternative solutions exist?

Manually adjusting the ClusterRole to remove the `node/proxy` RBAC permission.

### Additional context

This can be considered a breaking change and should not be included only in a new minor release + it should be highlighted in the release notes.

### PR DESCRIPTION
**Pull request by sigmabaryon**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind feature

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #937

**Please provide a short message that should be published in the vcluster release notes**
Remove the node/proxy permission from the cluster role when isolated mode is enabled.
Add a new Helm value to optionally allow original behavior even in the isolated mode.

**What else do we need to know?**

### CODE REVIEW

File: charts/eks/templates/rbac/clusterrole.yaml [hunk 1]
```diff
@@ -18,7 +18,9 @@ rules:
     resources: ["nodes", "nodes/status"]
     verbs: ["get", "watch", "list"]
   - apiGroups: [""]
-    resources: ["pods", "nodes/proxy", "nodes/metrics", "nodes/stats"]
+    resources: [ "pods",{{- if .Values.isolation.enabled }}
+      {{- if .Values.isolation.nodeProxyPermission.enabled}} "node/proxy",{{- end}}
+      {{- else}} "node/proxy",{{- end }} "nodes/metrics", "nodes/stats"]
```
Comment (matskiv) [inline charts/eks/templates/rbac/clusterrole.yaml:23]: I know the PR is still in draft state, but adding my 2cents to potentially save one review iteration :) 
I'd rather see `"nodes/proxy"` permission in a separate rule, which can then be wrapped in conditions. That would be more readable.

**Comment by matskiv** [comment]
@sigmabaryon Thank you for the PR. Could you please leave a comment on the related issue just so I can assign it to you? :)

File: charts/eks/templates/rbac/clusterrole.yaml (hunk 1 above)
Comment (sigmabaryon) [reply charts/eks/templates/rbac/clusterrole.yaml]: Have added a new rule for node/proxy permission for readability.

**Comment by matskiv** [comment]
I saw this error in the failing check for k3s:
```
 fatal  error executing helm upgrade vcluster ./charts/k3s --kubeconfig /tmp/1434243732 --namespace vcluster --install --repository-config='' --values /tmp/3549946504 --values ./test/commonValues.yaml --values ./test/e2e_node/values.yaml: Release "vcluster" does not exist. Installing it now.
Error: YAML parse error on vcluster/templates/rbac/clusterrole.yaml: error converting YAML to JSON: yaml: line 15: mapping values are not allowed in this context
```
Other charts probably have the same problem. You can check the "Create vcluster" step to see the error.

**Comment by sigmabaryon** [comment]
@matskiv I've fixed the issue related to the error reported in the tests.

**Comment by FabianKramm** [comment]
@sigmabaryon thanks, but it seems e2e tests are still failing, would you mind taking another look?

**Pull request merged**